### PR TITLE
02 - Added the desired deserialized elements of StoredGame.

### PR DESCRIPTION
**Pull request by xavierlepretre**

### CODE REVIEW

File: x/checkers/types/full_game.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	storedGame.Game = fullGame.Game.String()
+	storedGame.Turn = fullGame.Game.Turn.Color
+	storedGame.Red = fullGame.Red.String()
+	storedGame.Black = fullGame.Black.String()
+	return storedGame
+}
+
+func (storedGame *StoredGame) ToFullGame() (fullGame *FullGame) {
+	var err error
+	fullGame.Creator, err = sdk.AccAddressFromBech32(storedGame.Creator)
```
Comment (marbar3778) [inline x/checkers/types/full_game.go:28]: nit: no need for the var
```suggestion
	fullGame.Creator, err := sdk.AccAddressFromBech32(storedGame.Creator)
```

File: x/checkers/types/full_game.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	fullGame.Index = storedGame.Index
+	var game *rules.Game
+	game, err = rules.Parse(storedGame.Game)
```
Comment (marbar3778) [inline x/checkers/types/full_game.go:31]: err check

File: x/checkers/types/full_game.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
+package types
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/xavierlepretre/checkers/x/checkers/rules"
+)
+
+type FullGame struct {
```
Comment (marbar3778) [inline x/checkers/types/full_game.go:8]: what is the reason for having two types?

**Review by marbar3778 — APPROVED** [review]

File: x/checkers/types/full_game.go (hunk 3 above)
Comment (xavierlepretre) [reply x/checkers/types/full_game.go:8]: For instance, later on, like [here](https://github.com/cosmos/b9-checkers-academy-draft/pull/12/files#diff-5fec09d1b6ea66c9a28af8aad82729eb603eca996ead9d5d5691422473ba23c4R98), it allows us to use complex types, which make life easier.
Do you think I can achieve the same with extension methods of `StoredGame`?

**Pull request closed without merging**